targets/TARGET_TOSHIBA/TARGET_TMPM066/i2c_api.c

```diff
@@ -0,0 +1,361 @@
+        result = -1;
+    } else {
+        result = (int32_t)I2C_GetReceiveData(obj->i2c);
+    }
+    return (result);
+}
+
+void I2C_Start_Condition(i2c_t *p_obj, uint32_t data)
+{
+    status = I2C_GetState(p_obj->i2c);
+    if (status.Bit.BusState) {
+        I2C_SetRepeatStart(p_obj->i2c, ENABLE);
+    }
+    I2C_SetSendData(p_obj->i2c, (uint32_t)data);
+    I2C_GenerateStart(p_obj->i2c);
+}
+
+int i2c_byte_write(i2c_t *obj, int data)
+{
+    int32_t result;
+
+    I2C_ClearINTOutput(obj->i2c);
+
+    if(start_flag == 1) {
+        I2C_Start_Condition(obj, (uint32_t)data);
+        start_flag = 0;
+    } else {
+        I2C_SetSendData(obj->i2c, (uint32_t)data);
+    }
+
+    if (wait_status(obj) < 0) {
+        return (-1);
+    }
+
+    status = I2C_GetState(obj->i2c);
+    if (!status.Bit.LastRxBit) {
+        result = 1;
+    } else {
+        result = 0;
+    }
+    return (result);
+}
+
+
+#if DEVICE_I2CSLAVE
+
+#define I2C_NO_DATA             (0)
+#define I2C_READ_ADDRESSED      (1)
+#define I2C_WRITE_GENERAL       (2)
+#define I2C_WRITE_ADDRESSED     (3)
+
+void i2c_slave_mode(i2c_t *obj, int enable_slave)
+{
+    i2c_reset(obj);
+
+    obj->myi2c.I2CDataLen = I2C_DATA_LEN_8;
+    obj->myi2c.I2CACKState = ENABLE;
+    obj->myi2c.I2CClkDiv = clk.sck;
+    obj->myi2c.PrescalerClkDiv = clk.prsck;
+
+    if (enable_slave) {
+        obj->myi2c.I2CSelfAddr = obj->address;
+        I2C_SetINTI2CReq(obj->i2c, ENABLE);
+    } else {
+        obj->myi2c.I2CSelfAddr = SELF_ADDR;
+        NVIC_DisableIRQ(obj->IRQn);
+        I2C_ClearINTOutput(obj->i2c);
+    }
+    I2C_Init(obj->i2c, &obj->myi2c);
+}
+
+
+int i2c_slave_receive(i2c_t *obj)
+{
+    int32_t result = I2C_NO_DATA;
+
+    if  ((I2C_GetINTI2CStatus(obj->i2c)) && (I2C_GetSlaveAddrMatchState(obj->i2c))) {
+        status = I2C_GetState(obj->i2c);
+        if (!status.Bit.TRx) {
+            result = I2C_WRITE_ADDRESSED;
+        } else {
+            result = I2C_READ_ADDRESSED;
+        }
+    }
+    return (result);
+}
+
+int i2c_slave_read(i2c_t *obj, char *data, int length)
+{
+    int32_t count = 0;
+
+    while (count < length) {
+        int32_t pdata = i2c_byte_read(obj, ((count < (length - 1))? 0: 1));
+        status = I2C_GetState(obj->i2c);
+        if (status.Bit.TRx) {
+            return (count);
+        } else {
+            if (pdata < 0) {
+                break;
+            }
+            data[count++] = (uint8_t)pdata;
+        }
+    }
+    i2c_slave_mode(obj, 1);
+    return (count);
+}
+
+int i2c_slave_write(i2c_t *obj, const char *data, int length)
+{
+    int32_t count = 0;
+
+    while (count < length) {
+        if (i2c_byte_write(obj, (int32_t)data[count++]) < I2C_ACK) {
+            break;
+        }
+    }
+    i2c_slave_mode(obj, 1);
+    return (count);
+}
+
+void i2c_slave_address(i2c_t *obj, int idx, uint32_t address, uint32_t mask)
+{
+    obj->address = address & 0xFE;
+    i2c_slave_mode(obj, 1);
+}
+
+#endif
+
+void INTI2C0_IRQHandler(void)
```

Comment:
it's recommended to use nvic set /get vectors to register own handlers, than using weak symbols. 

Why are these empty?